test/e2e/module-layer/middleware.js

```diff
@@ -1,6 +1,7 @@
 import 'server-only'
 import * as React from 'react'
 import { NextResponse } from 'next/server'
+import { textValue } from './lib/mixed-lib/shared-module'
```

Comment:
Shouldn't this test that you'd get a client reference here instead of the actual value?  @timneutkens noticed that in middleware, with `react-server` you didn't get client references but probably should. But maybe that's for a follow-up

I'd usually opt for new tests instead of overloading existing ones to not accidentally reduce coverage in unrelated areas.

Reply:
shared-module is still server components here, the `textValue` is still the actual value. Unless you mark it as "use client". Only the `TestLink` here is the client reference.

Reply:
I separate the transform test cases into a new test suite

Reply:
Can you test with something that is obviously a client reference in this test e.g. create an explicit client module instead of relying on `next/link` being one.